.github/workflows/build.yml

```diff
@@ -1,33 +1,130 @@
 name: build
 on:
+  workflow_dispatch:
   push:
     branches:
       - main
+      - develop
   pull_request:
     branches:
       - main
       - develop
 jobs:
-  gradle:
+  build:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
 
       - uses: actions/setup-java@v4
         with:
-          distribution: temurin
+          distribution: corretto
           java-version: 21
-      - name: Setup Gradle
-        uses: gradle/gradle-build-action@v2
-      - name: Execute Gradle build
+
+      - uses: gradle/actions/setup-gradle@v3
+
+      - name: Build with Gradle
         run: ./gradlew build
 
       - uses: actions/upload-artifact@v4
         with:
-          name: Package
-          path: build/libs/*.jar
+          name: application
+          path: |
+            build/libs/jod-yksilo-*.jar
+            !build/libs/*-plain.jar
 
       - uses: actions/upload-artifact@v4
         with:
-          name: Reports
+          name: reports
           path: build/reports/
+
+  package:
+    needs: build
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ubuntu-latest
+    outputs:
+      tags: ${{ steps.docker-meta.outputs.tags }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: application
+          path: build/libs
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.JOD_UTIL_AWS_ROLE_ARN }}
```

Comment:
The role could be divided as follows:
```
role-to-assume: arn:aws:iam::${{ secrets.JOD_AWS_ACCOUNT_ID }}:role/${{ secrets.JOD_UTIL_AWS_ROLE_NAME}}
```

Reply:
Util role has a different account from the deploy role, so splitting would make configuration more complex, keeping as it is.